packages/git/src/browser/history/git-history-contribution.ts

```diff
@@ -53,30 +75,30 @@ export class GitHistoryContribution extends AbstractViewContribution<GitHistoryW
 
     registerCommands(commands: CommandRegistry): void {
         commands.registerCommand(GitHistoryCommands.OPEN_FILE_HISTORY, this.newUriAwareCommandHandler({
-            execute: async uri => {
-                const options: Git.Options.Log = {
-                    uri: uri.toString(),
-                    maxCount: GIT_HISTORY_MAX_COUNT,
-                    shortSha: true
-                };
-                this.showWidget(options);
-            }
+            execute: async uri => this.showWidget(uri.toString())
         }));
         commands.registerCommand(GitHistoryCommands.OPEN_BRANCH_HISTORY, {
-            execute: () => {
-                this.showWidget({
-                    maxCount: GIT_HISTORY_MAX_COUNT,
-                    shortSha: true
-                });
-            }
+            execute: () => this.showWidget(undefined)
         });
     }
 
-    async showWidget(options?: Git.Options.Log) {
+    async showWidget(uri: string | undefined) {
+        this.refreshWidget(uri);
+        this.openView({
+            activate: true
+        });
+    }
+
+    async refreshWidget(uri: string | undefined) {
```

Comment:
Does this need to be `public`?

Reply:
you are right. i will change it to protected